diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenWindowTraits.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenWindowTraits.kt
index 7af40fa..aa25575 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenWindowTraits.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/ScreenWindowTraits.kt
@@ -95,7 +95,7 @@ object ScreenWindowTraits {
         activity: Activity?,
         context: ReactContext?,
     ) {
-        if (activity == null || context == null) {
+        if (activity == null || context == null || EdgeToEdgeDetector.ENABLED) {
             return
         }
         if (defaultStatusBarColor == null) {
@@ -208,7 +208,7 @@ object ScreenWindowTraits {
         screen: Screen,
         activity: Activity?,
     ) {
-        if (activity == null) {
+        if (activity == null || EdgeToEdgeDetector.ENABLED) {
             return
         }
 
